cli/src/generate/templates/cmakelists.txt

```diff
@@ -0,0 +1,53 @@
+cmake_minimum_required(VERSION 3.13)
+
+project(tree-sitter-PARSER_NAME
+        VERSION "0.0.1"
+        DESCRIPTION "CAMEL_PARSER_NAME grammar for tree-sitter"
+        HOMEPAGE_URL "https://github.com/tree-sitter/tree-sitter-PARSER_NAME"
+        LANGUAGES C)
+
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
+set(TREE_SITTER_ABI_VERSION ABI_VERSION_MAX CACHE STRING "Tree-sitter ABI version")
+if(NOT ${TREE_SITTER_ABI_VERSION} MATCHES "^[0-9]+$")
+    unset(TREE_SITTER_ABI_VERSION CACHE)
+    message(FATAL_ERROR "TREE_SITTER_ABI_VERSION must be an integer")
+endif(NOT ${TREE_SITTER_ABI_VERSION} MATCHES "^[0-9]+$")
+
+find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
+
+add_custom_target(generate DEPENDS src/parser.c)
+add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
+                   DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
+                   COMMAND "${TREE_SITTER_CLI}" generate --no-bindings
+                           --abi=${TREE_SITTER_ABI_VERSION} src/grammar.json
+                   WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+                   COMMENT "Generating parser.c")
+
+add_library(tree-sitter-PARSER_NAME src/parser.c)
+if(EXISTS src/scanner.c)
+  target_sources(tree-sitter-PARSER_NAME PRIVATE src/scanner.c)
+endif(EXISTS src/scanner.c)
+target_include_directories(tree-sitter-PARSER_NAME PRIVATE src)
+
+set_target_properties(tree-sitter-PARSER_NAME
+                      PROPERTIES
+                      C_STANDARD 11
+                      POSITION_INDEPENDENT_CODE ON
+                      SOVERSION "${PROJECT_VERSION_MAJOR}.${TREE_SITTER_ABI_VERSION}")
```

Comment:
Shall we bikeshed on whether the TS ABI version should be the major or the minor?